src/Formless/Component.purs

```diff
@@ -321,6 +323,12 @@ handleAction handleAction' handleEvent action = flip match action
       _ <- handleAction handleAction' handleEvent FA.validateAll
       IC.submit >>= traverse_ (Submitted >>> handleEvent)
 
+  , submitPreventDefault: \event -> do
+      H.liftEffect $ Event.preventDefault event
+      _ <- IC.preSubmit
+      _ <- handleAction handleAction' handleEvent FA.validateAll
+      IC.submit >>= traverse_ (Submitted >>> handleEvent)
```

Comment:
First major change to review

Reply:
Can we reuse `submit` here by calling it recursively? We'd call `preventDefault` on the event, and then call `submit` after that.

Reply:
I tried the following (which may not be what you were envisioning),
```
...
  , submitPreventDefault: \event -> do      
      H.liftEffect $ Event.preventDefault event      
      handleAction handleAction' handleEvent submit
...
where submit = inj (Proxy :: _ "submit") unit
```
But this just injects another event because submit doesn't take an event. I also tried pulling out the submit logic into a where clause but the type errors are really unwieldy.

Reply:
^ ignore this. I didn't bundle the app correctly